### ISSUE TITLE
When $sift() results in an empty object, the empty object should be returned.

### ISSUE THREAD

**Issue by jhorbulyk** [jsonata-js/jsonata#339]
Consider the following input:
`{}`
and the following expression:
`$sift($, function($v,$k){true})`
The expected result would be
`{}`
however it is instead
*no match*

**Comment by andrew-coleman** [comment]
That's true, but it's done for a reason, and there is [explicit code](https://github.com/jsonata-js/jsonata/blob/master/src/functions.js#L1766) that makes this happen.  Consider [this example](http://try.jsonata.org/rybDAjPlH) that extracts only the objects that contain a `Postcode` property. If we were to remove the 'empty object' check, then we would end up with the following result:
```
[
  {
    "Address": {
      "Street": "Hursley Park",
      "City": "Winchester",
      "Postcode": "SO21 2JN"
    }
  },
  {},
  {},
  {},
  {},
  {},
  {},
  {},
  {
    "Alternative.Address": {
      "Street": "Brick Lane",
      "City": "London",
      "Postcode": "E1 6RF"
    }
  },
  {}
]
```

**Comment by jhorbulyk** [comment]
Perhaps some sort of flag could solve this problem?

**Comment by andrew-coleman** [comment]
Could you elaborate on what sort of flag you are proposing, and what its effect would be, given the above discussion.  Thanks.

**Comment by jhorbulyk** [comment]
> Could you elaborate on what sort of flag you are proposing, and what its effect would be, given the above discussion. 

Modify signature of $sift() to `$sift(object, function, returnEmptyObjects)`.  `returnEmptyObjects` is a boolen that defaults to `false`.  When the result of a sift operation is equal to `{}` and `returnEmptyObjects` is true, then `{}` is returned.  Otherwise `undefined` is returned.
